include/picongpu/particles/atomicPhysics/electronDistribution/histogram2/AdaptiveHistogram.hpp

```diff
@@ -0,0 +1,722 @@
+/* Copyright 2020-2021 Brian Marre
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file This file implements a adaptive Histogram starting from argument 0
+ *
+ * basic idea: 1-dimensional infinite histogram, with sparsely populated bins
+ * binWidths are not uniform, but indirectly definded by a target value of a
+ * relative binning width dependent error.
+ *
+ * only populated bins are stored in memory.
+ *
+ * TemplateParameters:
+ * -------------------
+ *  @tparam T_AtomicDataBox ... type of data container for atomic input data
```

Comment:
These parameters should be doxygen comments over the class/struct definition.

Reply:
done